src/main/java/com/github/cukespace/arquillian/asciidoctor/AsciidoctorObserver.java

```diff
@@ -43,19 +44,54 @@
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
+import java.util.concurrent.Executors;
 import java.util.logging.Logger;
 import java.util.regex.Pattern;
 
 import static java.util.Arrays.asList;
 
 // fork of asciidoctor maven plugin
 public class AsciidoctorObserver {
+    
     private static final Pattern ASCIIDOC_EXTENSION_PATTERN = Pattern.compile("^[^_.].*\\.a((sc(iidoc)?)|d(oc)?)$");
+    
+    private static Asciidoctor asciidoctor;
 
     @Inject
     private Instance<ArquillianDescriptor> descriptorInstance;
 
-    public void init(@Observes final EventContext<ManagerStopping> ending) {
+    public void start(@Observes final EventContext<ManagerStarted> starting) {
+        starting.proceed();
+        final ArquillianDescriptor descriptor = descriptorInstance.get();
+        Executors.newSingleThreadExecutor().execute(new Runnable() {
+            
+            @Override
+            public void run() {
+                initAsciidoctor(descriptor);   
+            }
+        });
+    }
+    
+    private void initAsciidoctor(ArquillianDescriptor arquillianDescriptor) {
+
+        String gemPath = null;
+        for (final ExtensionDef extensionDef : arquillianDescriptor.getExtensions()) {
```

Comment:
don't you think we can have only one adoc instance? gemPath can change between different configurations? do you have an example where current impl would break?